app/src/main/java/com/practicum/playlistmaker/library/data/db/FavoriteTrackRepositoryImpl.kt

```diff
@@ -0,0 +1,42 @@
+package com.practicum.playlistmaker.library.data.db
+
+import com.practicum.playlistmaker.library.data.db.converters.FavoriteTrackDbConvertor
+import com.practicum.playlistmaker.library.data.db.entity.FavoriteTrackEntity
+import com.practicum.playlistmaker.library.domain.db.FavoriteTrackRepository
+import com.practicum.playlistmaker.search.domain.models.Track
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class FavoriteTrackRepositoryImpl(
+    private val appDatabase: AppDatabase,
+    private val favoriteTrackDbConvertor: FavoriteTrackDbConvertor
+) : FavoriteTrackRepository {
+
+    override suspend fun insertFavoriteTrack(track: Track?) {
```

Comment:
Вряд ли есть какой-то смысл вставлять нуллабельный трек:
```suggestion
    override suspend fun insertFavoriteTrack(track: Track) {
```
Проверку на null лучше осуществлять до вставки